org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/wizard/welcome/WelcomePageConfiguratorFactory.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2015 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Simon Scholz <simon.scholz@vogella.com> - Bug 473545
+ */
+
+package org.eclipse.buildship.ui.wizard.welcome;
+
+import java.util.ArrayList;
+
+import org.eclipse.buildship.ui.wizard.project.ProjectWizardMessages;
+import org.eclipse.buildship.ui.wizard.welcome.internal.WelcomePageConfiguratorImpl;
+
+/**
+ * Factory for creating {@link WelcomePageConfigurator} instances for the
+ * {@link GradleWelcomeWizardPage}.
+ */
+public class WelcomePageConfiguratorFactory {
+
+    private WelcomePageConfiguratorFactory() {
+    }
+
+    public static WelcomePageConfigurator createImportWizardWelcomePageConfigurator(
+            String welcomePageEnabledPrefernceKey) {
+        ArrayList<WelcomePageParagraph> paragraphs = new ArrayList<WelcomePageParagraph>(3);
+        paragraphs.add(new WelcomePageParagraph(
+                ProjectWizardMessages.WelcomePageConfiguratorFactory_Import_Wizard_Paragraph_Title1,
+                ProjectWizardMessages.WelcomePageConfiguratorFactory_Import_Wizard_Paragraph_Content1));
+        paragraphs.add(new WelcomePageParagraph(
```

Comment:
We never use the class name as part of the i18n keys. Can we please keep it that way?
Also, can we please use specific names rather than generic ones? e.g. something like Title_ImportWizard_Distribution